---
title: "Data science"
format: 
  html:
    toc: true
    toc-depth: 3
    toc-location: left
    number-sections: true
editor: visual
---

Despite their ecological importance, efforts to catalog protistan physiological abilities and trophic strategies lag behind those of their prokaryotic counterparts. The interpretation of large sequence datasets relies on the knowledge gleaned from culture-based transcriptome studies, new developments in genetic probing of microorganisms, and single-cell sequencing. It is imperative that we build reproducible pipelines and infrastructure to keep up with growing sequence information.

# Computational biology
::: column-margin
![](images/Hu Lab Logo Final-01.png)
:::
## Tag-sequencing

To address questions related to microeukaryotic biodiversity, we use 18S rRNA gene tag-sequencing (also called metabarcoding). Below are resources for analyzing this type of data. 

Our most recent [tag-sequencing work flow](https://github.com/shu251/qiime2-2024) was updated in 2024. 

Check my [github](https://github.com/shu251) for _in process_ work, where we are always working to update our code. 

Also see the [BVCN](https://biovcnet.github.io) resources for amplicon sequencing

## Time-series data

When we work with any metabarcoding data, it is important to acknowledge and analyze it as _compositional_ data. As part of our R code clubs we always emphasize the inclusion of diagnostic steps to make sure our data is being analyzed appropriately. 

A product from a series of discussions with colleagues is [resource for how to analyze time-series data.](https://github.com/shu251/analyzing_microbiome_timeseries). When paired with the [publication](https://www.frontiersin.org/journals/genetics/articles/10.3389/fgene.2020.00310/full) it is a good reminder and starting point when analyzing this type of dataset. 

Additional resources include the [code](https://github.com/shu251/18Sdiversity_diel) and publication, called [A Hard Day's Night](https://www.frontiersin.org/articles/10.3389/fmars.2018.00351/full), which explores temporal niche partitioning among Station ALOHA microeukaryotic communities. 


![Figure 2 in Coenen et al. 2020](images/timeseries.jpg){.lightbox fig-align="left" width="300"}

## Metatranscriptomics

To interrogate _in situ_ microeukaryotic metabolic activity, we can target the messengar RNA (mRNA) for sequencing through metatranscriptomics. 

Our most recent [introductory tutorial on metatranscriptomics, especially for eukaryotes, can be found here](https://shu251.github.io/meta-eukomic/main.html). 

### Computational tools

An active area of our research includes developing solutions to the computational challenges presented by eukaryotic genetics, a part of this involves creating tools to automate the assembly, annotation, and processing of metaâ€™omic (metagenome/metatranscriptome) data (EUKulele, EUKHeist) and augment how sequence reference databases are queried (see Krinos et al. 2024).


[EUKHeist code](https://github.com/alexanderlabwhoi/EukHeist) and [publication](https://journals.asm.org/doi/full/10.1128/mbio.01676-23?rfr_dat=cr_pub++0pubmed&url_ver=Z39.88-2003&rfr_id=ori%3Arid%3Acrossref.org)

::: column-margin
![Alexander et al. 2023](images/eukheist.jpg)
:::

[EUKulele documentation](https://eukulele.readthedocs.io/en/latest/) and [publication](https://joss.theoj.org/papers/10.21105/joss.02817).

### Building reproducible pipeline for amplicon sequence analysis

Instructions for implementing a snakemake workflow to make tag-sequencing pipeline more robust [HERE](https://github.com/shu251/tagseq-qiime2-snakemake)

[Building/prepping](https://github.com/shu251/db-build-microeuks) a database for use in QIIME2

### Metaeukomic

Members of the Hu lab are contributing to community-wide efforts to examine the range of variability across metatranscriptome bench and computational pipelines. This an intercalibration effort funded by OCB, [Metaeukomic](https://www.us-ocb.org/metat-intercomparison/). 

# Database Darkly

::: column-margin
![](images/db_darkly_logo_emptybg.png)
:::

A challenge we face with deep-sea biodiversity work is the lack of available genetic reference databases. Using results from a [large tag-sequencing survey](https://onlinelibrary.wiley.com/doi/abs/10.1111/mec.16745), a team of undergraduate researchers took a deep dive into mining biological and ecological information about deep-sea protistan species.

[Visit the Database Darkly website](https://shu251.github.io/deep-sea-microeuk-DB/) to learn more about the microeukaryotes that live in the dark ocean.

![](images/OCNG-symposium-UndergraduateResearch.png){.lightbox,fig-align="center", width="450"}

# R tutorials & workshops
-   [R code examples and tutorials](https://shu251.github.io/r-ventures/#introduction)

-   [Generating figures](https://github.com/shu251/PreliminaryFigures_V4_tagseq) from HTS tag-sequencing data - OTU or ASV table as input

-   Learning R with [BVCN](https://biovcnet.github.io)


-   [Extracting microbial eukaryotic information from metagenomes](https://github.com/shu251/Extract_microeukrRNA_from_metagenome)


# R packages & builds

## Axial Seamount Wordle

[Play our custom Axial Seamount-themed wordle](https://sarah-k-hu.shinyapps.io/wordle-axial/)

Part of our [#protatax](https://www2.whoi.edu/staff/jhuber/protatax2022/) expedition.

```{r out.width ="50%", echo=FALSE}
knitr::include_graphics("images/vent-logo-protatax_JulieForCruise.jpg")
```



## Woods Hole color palette

Download color palettes inspired by Cape Cod and Woods Hole area. Includes docks, sunsets, WHOI logo, and RV Atlantis. [github](https://github.com/shu251/PaletteWoodsHole)

```{r out.width ="50%", echo=FALSE}
knitr::include_graphics("images/knob1.png")

```


## ExeRcise

Generate a randomized custom workout. Inspired by High Intensity Interval training workouts, user defines warm up and workout interval length and total number of exercises. Additional functions to run workout with a timer in R and a tabata-style workout. [github](https://github.com/shu251/exRcise)

------------------------------------------------------------------------

## Lab work

Built *for personal use only*. List of functions to streamline molecular bench work and common math required in preparing genetic material for sequencing. e.g., PCR master mix calculation, CV=CV, and RNA quantification preparation steps. [github repo](https://github.com/shu251/lab)



```{r out.width="50%", echo=FALSE}
knitr::include_graphics("images/knob.jpg")
```

*the Knob in Woods Hole, MA*