---
title: "Resources"
description: |
  Links for resources to learn code
output: 
  distill::distill_article:
    css: theme.css
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE)
```

# Meta'omics

* [EUKHeist](https://github.com/alexanderlabwhoi/EukHeist) & [preprint] (https://www.biorxiv.org/content/10.1101/2021.07.25.453713v1)

* [EUKulele](https://eukulele.readthedocs.io/en/latest/) 

* [Code](https://github.com/shu251/SPOT_metatranscriptome) from [a metatranscriptomic survey of protistan trophic strategies](https://onlinelibrary.wiley.com/doi/abs/10.1111/1462-2920.14259)
  + [EukZoo](https://zenodo.org/record/1476236#.XRdsuC2ZNTY) database, an aquatic protistan protein database for meta-omics studies

# Amplicon sequence analysis

* I am currently working on streamlined pipelines to determine Amplicon Sequence Variants (ASVs) or Operational Taxonomic Units (OTUs) using **QIIME2** or **DADA2** in R for use in environmental microbial ecology.

# Data management

* [Keeping my R life organized](https://alexanderlabwhoi.github.io/post/anaconda-r-sarah/) details how I set up R with anaconda on my computer. Introduction to managing projects with conda environments. Includes description of R, RStudio, and IRKernel (jupyter notebooks).

# Introduction to R

* [R code examples and tutorials](https://shu251.github.io/r-ventures/#introduction)

* [Generating figures](https://github.com/shu251/PreliminaryFigures_V4_tagseq) from HTS tag-sequencing data - OTU or ASV table as input

* Learning R with [BVCN](https://biovcnet.github.io)


# Practical Ecology

* Resources for time-series analysis: [code and analyses](https://github.com/shu251/18Sdiversity_diel) from [A Hard Day's Night](https://www.frontiersin.org/articles/10.3389/fmars.2018.00351/full)


## Introduction to tag-sequencing 

* [QIIME1](https://github.com/shu251/V4_tagsequencing_18Sdiversity_q1) or 
* [QIIME2](https://github.com/shu251/qiime2_ASVworkflow_v8).
* Also see the [BVCN](https://biovcnet.github.io) resources for amplicon sequencing

## Building reproducible pipeline for amplicon sequence analysis

* Implementing a snakemake workflow to make tag-sequencing pipeline more robust [HERE](https://github.com/shu251/tagseq-qiime2-snakemake)
  + [Building/prepping](https://github.com/shu251/db-build-microeuks) a database for use in QIIME2
