<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.1.251">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>Main themes</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1.6em;
  vertical-align: middle;
}
</style>


<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit"
  }
}</script>


</head>

<body class="nav-sidebar docked fullcontent">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
  <nav class="quarto-secondary-nav" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
    <div class="container-fluid d-flex justify-content-between">
      <h1 class="quarto-secondary-nav-title">Main themes</h1>
      <button type="button" class="quarto-btn-toggle btn" aria-label="Show secondary navigation">
        <i class="bi bi-chevron-right"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse sidebar-navigation docked overflow-auto">
    <div class="pt-lg-2 mt-2 text-left sidebar-header">
      <a href="./" class="sidebar-logo-link">
      <img src="./images/Hu Lab Logo Final-01.png" alt="" class="sidebar-logo py-0 d-lg-inline d-none">
      </a>
      <div class="sidebar-tools-main tools-wide">
    <a href="https://twitter.com/shu251" title="" class="sidebar-tool px-1"><i class="bi bi-twitter"></i></a>
    <a href="https://github.com/shu251" title="" class="sidebar-tool px-1"><i class="bi bi-github"></i></a>
    <a href="https://scholar.google.com/citations?user=9CI91CMAAAAJ&amp;hl=en" title="" class="sidebar-tool px-1"><i class="bi bi-scholar"></i></a>
</div>
      </div>
      <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
      </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./index.html" class="sidebar-item-text sidebar-link">Home</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./team.html" class="sidebar-item-text sidebar-link">Lab group</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./publications.html" class="sidebar-item-text sidebar-link">Publications</a>
  </div>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" aria-expanded="true">Research</a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" aria-expanded="true">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./projects.html" class="sidebar-item-text sidebar-link active">Main themes</a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" aria-expanded="true">Resources</a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" aria-expanded="true">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-2" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./resources.html" class="sidebar-item-text sidebar-link">resources</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./r-packages.html" class="sidebar-item-text sidebar-link">R packages &amp; projects</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./lab-policies.html" class="sidebar-item-text sidebar-link">Lab policies</a>
  </div>
</li>
      </ul>
  </li>
    </ul>
    </div>
</nav>
<!-- margin-sidebar -->
    
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title d-none d-lg-block">Main themes</h1>
</div>



<div class="quarto-title-meta">

    
    
  </div>
  

</header>

<section id="interconnected-microbial-eukaryome" class="level1">
<h1>Interconnected microbial eukaryome</h1>
<p>The <em>interconnected microbial eukaryome</em> means I am interested in the relationship between microbial eukaryotes and their environment, and microbial eukaryotes and other organisms. Research typically starts with questions such as <em>Who is present?</em>, <em>What are they doing?</em>, and <em>What is their ecological role in marine environments?</em></p>
<p>Protistan species make up the majority of the branches on the eukaryotic tree of life. Similar to their vast and complex morphological ( <em>e.g.</em>, skeletal structure, modes of mobility, and size), the nutritional strategies (or feeding methods) protists are capable of are diverse and complicated. The three core <em>trophic modes</em> include <strong>phototrophy</strong>, <strong>heterotrophy</strong>, and <strong>mixotrophy</strong>. These trophic strategies place protists at the center of marine food webs – meaning, they mediate or facilitate central routes of energy and nutrient flux. Hence, protists play several key ecological roles at the base of marine food webs.</p>
<section id="protists" class="level2">
<h2 class="anchored" data-anchor-id="protists">Protists</h2>
<p>Single-celled microbial eukaryotes <em>or protists</em> are critical team players in most ecosystems on our planet. Protistan communities are composed of hundreds to thousands of species. The species composition of protistan communities can serve as a sentinel of the surrounding environment.</p>
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/Screen%20Shot%202021-09-21%20at%2012.38.40%20PM.png" class="img-fluid figure-img" width="1079"></p>
</figure>
</div>
</div>
</div>
<section id="three-core-research-initiatives" class="level3">
<h3 class="anchored" data-anchor-id="three-core-research-initiatives">Three core research initiatives</h3>
<p>From coastal environments to deep-sea habitats, protistan community structure (diversity and composition of species) and nutritional strategies are reflections of their environment. While protists form a central theme in my work, my goal is to make sense of how microorganisms fuel and facilitate the processes sustaining our planet. Through the study of microbial eukaryotes, we learn how diverse nutritional strategies of protists are tied to season, time of day, and availability of resources. My main goal is to understand how protistan communities interact with other microorganisms and their environment to support marine ecosystems.</p>
<p>Three core research themes:</p>
<ul>
<li><p><strong>Meta’omics to resolve taxon-specific ecological roles</strong> - Leverage temporal &amp; spatial scales to link genotype &amp; phenotype in natural protistan assemblages.</p></li>
<li><p><strong>Trophic interactions at hydrothermal vent food webs</strong> - Heterotrophic microbial eukaryotes associated with deep-sea vents play an important role in deep-sea carbon cycling.</p></li>
<li><p><strong>Data science</strong> - Computational biology is a wonderful opportunity for interdisciplinary collaborations. In addition to using data science, I aim to build a “data science community”</p></li>
</ul>
<hr>
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="projects_files/figure-html/unnamed-chunk-2-1.png" class="img-fluid figure-img" width="768"></p>
</figure>
</div>
</div>
</div>
</section>
</section>
</section>
<section id="deep-sea-hydrothermal-vents" class="level1">
<h1>Deep-sea hydrothermal vents</h1>
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/IMG_3510-Edit-Edit.jpg" class="img-fluid figure-img" style="width:40.0%"></p>
<p></p><figcaption class="figure-caption">Mid-Cayman Rise. Credit: WHOI</figcaption><p></p>
</figure>
</div>
</div>
</div>
<p>Highly reduced and thermally charged venting fluids from the subseafloor mix with surrounding seawater, creating a sharp geochemical gradient which promotes a hub of biological diversity at the site of venting fluid. Studies of the bacterial and archaeal chemosynthetic populations at hydrothermal vent sites have highlighted the important roles these microorganisms play in deep-sea carbon cycling and offered a unique window into subseafloor microbiology. To date, studies of deep-sea food webs and ecological interactions do not typically cover the microbial eukaryotic assemblages.</p>
<p>Heterotrophic protists are ubiquitous in all aquatic ecosystems and represent an important ecological link by transferring organic carbon from primary producers to higher trophic levels. Previous work has shown protists to be present and active at deep-sea hydrothermal vents and efforts to characterize the composition of these populations has revealed many species to be heterotrophic grazers and parasites. However, efforts to quantify protistan grazing pressure in the deep sea are rare.</p>
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/foodweb-map-simple.png" style="height:2.0%" width="490" class="figure-img"></p>
<p></p><figcaption class="figure-caption">Simplified food web</figcaption><p></p>
</figure>
</div>
</div>
</div>
<p>Supported by the <a href="https://www.darkenergybiosphere.org/">Center for Dark Energy Biosphere Investigations STC</a>, I joined <a href="https://www2.whoi.edu/staff/jhuber/projects/">Dr.&nbsp;Julie Huber’s lab</a> at <a href="https://www.whoi.edu/">WHOI</a>. Since 2018, I have been involved in several expeditions to hydrothermal vents (see below) to address these main questions:</p>
<blockquote class="blockquote">
<p>What is the role of protists in deep-sea hydrothermal vent food web ecology?</p>
</blockquote>
<blockquote class="blockquote">
<p>How does protistan diversity, distribution, and activity influence carbon flux in the deep sea?</p>
</blockquote>
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/cdebi-logo.png" class="img-fluid figure-img" style="width:20.0%"></p>
</figure>
</div>
</div>
</div>
<section id="accessing-deep-sea-food-webs" class="level2">
<h2 class="anchored" data-anchor-id="accessing-deep-sea-food-webs">Accessing deep-sea food webs</h2>
<p>To assess microbial biodiversity, we rely on sequencing the DNA from microbial cells that are collected onto a filter. The genetic material is extracted and sequenced so we can gain a snapshot of the microbial community composition <em>(who is present?)</em>. These molecular approaches have been critical in gaining the knowledge we currently have on the microbial communities in deep-sea and difficult to range habitats. Actually getting to deep-sea habitats takes a lot of engineering and planning. Broadly, we can use remotely- or human-operated vehicles (ROVs and HOVs) from a ship. Often we have to run experiments on the ship once the ROV returns with hydrothermal vent fluid or have the ROV collect samples <em>in situ</em>.</p>
<p>For this work, I run incubations with diffuse flow fluid collected via ROV to capture the grazing activity of heterotrophic protists. Consumption of microbial prey by protistan grazers (heterotrophs) is a key route of carbon exchange (the transfer of chemosynthetic microorganisms to higher trophic level), and these experiments allow us to obtain a grazing rate (cells consumed ml^-1 day ^day). In order to assess the level of grazing pressure by protists, I compare and contrast the grazing rates and community composisions of protistan assemblages within discharging vent fluid, the plume (hydrothermally-influenced environment above the vent site), and the background deep-sea water.</p>
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/vent-with-location.png" class="img-fluid figure-img" style="width:30.0%"></p>
</figure>
</div>
</div>
</div>
<p>In aquatic habitats, transition zones driven by changes in chemistry or nutrients can create biological ‘hotspots’ of microbial activity. Within the hydrothermal vent environment</p>
</section>
<section id="microbes-need-frenemies" class="level2">
<h2 class="anchored" data-anchor-id="microbes-need-frenemies">Microbes need frenemies</h2>
<p>Can the interaction of microbes with others be defined as friendly? rival? or <em>frenemy</em>?</p>
<p>Ecological interactions among microbes (bacteria and archaea), viruses, and eukaryotic microorganisms are critical junctions in marine food webs. These interactions range from mutually beneficial relationships to sources of microbial mortality. Interactions between viruses-microbes and eukaryotes-microbes at deep-sea hydrothermal vents impact local carbon cycling. This project aims to identify these microbial interactions, specifically those related to cell death by protistan grazing or viral lysis, and explore how they vary across different hydrothermal vent habitats. By providing a better understanding of the composition and nature of these relationships, the investigators aim to build a better food web model of deep-sea hydrothermal vents and improve our understanding of how climate change and other human activities impact the ecosystem.</p>
<p>Outcomes from this project include the generation of new microbiology, oceanography, and computer science curricula targeted at community college students. In addition, it involves research with undergraduate students at all stages of the research process and provides opportunities for professional development and peer-to-peer mentoring.</p>
<p><a href="https://www.bco-dmo.org/project/875053">Newly awarded NSF proposal with Rika Anderson &amp; Julie Huber</a></p>
<p><em>Stay tuned for opportunities!</em></p>
<section id="field-work" class="level3">
<h3 class="anchored" data-anchor-id="field-work">Field work</h3>
<section id="axial-seamount" class="level4">
<h4 class="anchored" data-anchor-id="axial-seamount">Axial Seamount</h4>
<p>We are currently planning work at Axial Seamount (NE Pacific Ocean) as part of an <a href="https://nsf.gov/awardsearch/showAward?AWD_ID=1947776&amp;HistoricalAwards=false">NSF-funded proposal</a> to characterize the rate and route of carbon via phagotrophic protists.</p>
<p>Axial Seamount is an active submarine volcano on the Juan de Fuca Ridge in the NE Pacific Ocean, off the coast of Oregon. The microbiome of the low-temperature (&lt;100C) diffuse vent sites in the region have been frequently studied and the vent field within the caldera include a range of distinct geochemistries.</p>
<hr>
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/jason-van.jpg" class="img-fluid figure-img" style="width:50.0%"></p>
<p></p><figcaption class="figure-caption">Jason van on the RV Atlantis</figcaption><p></p>
</figure>
</div>
</div>
</div>
</section>
<section id="mid-cayman-rise---2020-expedition" class="level4">
<h4 class="anchored" data-anchor-id="mid-cayman-rise---2020-expedition">Mid-Cayman Rise - 2020 Expedition</h4>
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/jason-sarah.png" class="img-fluid figure-img" style="width:25.0%"></p>
<p></p><figcaption class="figure-caption">ROV Jason during the Mid-Cayman Rise cruise</figcaption><p></p>
</figure>
</div>
</div>
</div>
<p>I was a part of the RV Atlantis (AT42-22) research cruise to study the geochemistry and microbiology of the Von Damm and Piccard vent fields, located on the Mid-Cayman Rise. We completed 9 dives with ROV Jason at the Von Damm and Piccard vent fields. Hydrothermal vent fluid collection was facilitated by the isobaric gas-tight (IGT) fluid samplers (9) and the hydrothermal organic geochemistry (HOG) sampler (Susan Lang). CTD-rosette casts were also conducted to obtain water column plume and background seawater. Relevant objectives included (1) conducting grazing incubations and (2) collecting samples from the hydrothermal vent, plume, and background seawater environments. Laboratory work for analyzing the grazing experiments and cell counts has been completed and the molecular work is in progress.</p>
<p>Samples from Von Damm (10 sites) and Piccard (8 sites) vent fields will provide detailed analysis of the microbial composition, biogeography, and role in the food web. 3 of the grazing experiments originated from CTD-rosette casts from the background and plume environment and 6 were conducted with fluid from the HOG. 4 grazing experiments were performed with the IGT, so incubations remained at <em>in situ</em> pressure.</p>
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/J2-618-IGT1%20-%20Beebe%20Woods.jpg" class="img-fluid figure-img" style="width:30.0%"></p>
<p></p><figcaption class="figure-caption">Beebe Woods vent at Mid-Cayman Rise. Credit: WHOI</figcaption><p></p>
</figure>
</div>
</div>
</div>
<hr>
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/cam2_20190607132644_edited.jpg" class="img-fluid figure-img" style="width:100.0%"></p>
<p></p><figcaption class="figure-caption">ROV Hercules at newly discovered Apollo vent field. Credit: OET/Nautilus Live</figcaption><p></p>
</figure>
</div>
</div>
</div>
</section>
<section id="gorda-ridge---2019-expedition" class="level4">
<h4 class="anchored" data-anchor-id="gorda-ridge---2019-expedition">Gorda Ridge - 2019 Expedition</h4>
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/Sarah-and-Amy-Controlvan-2.jpg" class="img-fluid figure-img" style="width:30.0%"></p>
<p></p><figcaption class="figure-caption">In the ROV Hercules control van with Amy Smith. Credit: OET/Nautilus Live</figcaption><p></p>
</figure>
</div>
</div>
</div>
<p>Sea Cliff and Apollo vent fields along the Gorda Ridge spreading center, located ~200 km off the coast of southern Oregon, were visited in May-June 2019 with the E/V Nautilus (cruise NA108; 5). Low temperature diffuse hydrothermal vent fluid samples &lt;100ºC were collected using the ROV Hercules and a SUspended Particle Rosette Sampler (SUPR; 6). The overarching goal of the cruise was to integrate scientific investigation of the deep sea with the exploration of ocean worlds on other planets, as the deep-sea hydrothermal vent system serves as an analog environment (a part of the SUBSEA program (Systematic Underwater Biogeochemical Science and Exploration Analog)).</p>
<p>Findings from the Gorda Ridge cruise are published: Hu, S.K., Herrera, E.L., Smith, A.R., Pachiadaki, M.G., Edgcomb, V.P., Sylva, S.P., Chan E.W., Seewald, J.S., German, C.R., &amp; Huber, J.A. (2021) Protistan grazing impacts microbial communities and carbon cycling in the deep-sea hydrothermal vent environment. Proc Natl Acad Sci USA <a href="https://www.pnas.org/content/118/29/e2102674118">link</a> <a href="https://shu251.github.io/protist-gordaridge-2021/">code</a></p>
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/IMG_4779.jpg" class="img-fluid figure-img" style="width:20.0%"></p>
<p></p><figcaption class="figure-caption">Sampling microcolonizer experiments on board the EV Nautilus. Credit: OET/Nautilus Live</figcaption><p></p>
</figure>
</div>
</div>
</div>
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/mtedwards-cones.png" class="img-fluid figure-img" style="width:60.0%"></p>
<p></p><figcaption class="figure-caption">Mt Edwards vent site at Gorda Ridge with microcolonizer experiments. Credit: OET/Nautilus Live</figcaption><p></p>
</figure>
</div>
</div>
</div>
<p>All 6 microcolonizer experiments (pictured above) were placed on and near active diffuse flow at Mt. Edwards, each had substrates that included: shell (CaCO3) Riftia tubeworm shell (chitin), quartz, pyrite, basalt, and olivine). Microcolonizers experienced a range of temperatures, influenced by the diffusely venting fluid nearby</p>
<ul>
<li><p>What species colonize these substrates first?</p></li>
<li><p>Does species richness and evenness vary by substrate type?</p></li>
<li><p>Are the same species found on the substrates also found at the diffuse vent fluid?</p></li>
</ul>
<p>A summary of my findings were presented as part of the ISOP virtual poster session.</p>
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/GR-comic-protistsession-updated.png" class="img-fluid figure-img" style="width:80.0%"></p>
<p></p><figcaption class="figure-caption">Graphic comic poster for ISOP, August 2020</figcaption><p></p>
</figure>
</div>
</div>
</div>
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="projects_files/figure-html/unnamed-chunk-15-1.png" class="img-fluid figure-img" width="768"></p>
</figure>
</div>
</div>
</div>
</section>
</section>
</section>
</section>
<section id="time-series" class="level1">
<h1>Time-series</h1>
<p>The diversity of microbial eukaryotic trophic strategies makes them key players in marine food webs. Protists are single-celled microbial eukaryotes that play numerous ecological roles at the epicenter of marine food webs as primary producers, consumers, mixotrophs (mixed phototrophic and heterotrophic nutrition), and as partners in various mutualistic or parasitic associations. Within the diverse array of eukaryotic species, protist functional groups that exhibit heterotrophic nutrition or combined phototrophic and heterotrophic nutrition (mixotrophy) have been especially overlooked, especially from in situ studies.</p>
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/SPOT-fieldwork.png" class="img-fluid figure-img" style="width:40.0%"></p>
<p></p><figcaption class="figure-caption">SPOT cruise</figcaption><p></p>
</figure>
</div>
</div>
</div>
<section id="the-spot-station" class="level2">
<h2 class="anchored" data-anchor-id="the-spot-station">the SPOT station</h2>
<p>The <em>S</em>an <em>P</em>edro <em>O</em>cean <em>T</em>ime-series (SPOT) station is located about half way between the Port of Los Angeles and Santa Catalina Island. <a href="https://dornsife.usc.edu/spot/">SPOT</a> is a part of the Wrigley Institute and University of Southern California, it is a long-term study site that has been visited, sampled, and studied monthly for over 15 years. As a PhD student, one of my main roles was sampling every month - which meant that in addition to contributing to a long-term sampling effort, I got to ask my own research questions that were scientifically and logistically support by this station.</p>
<p>Each month, we would conduct CTD casts to obtain oceanographic measurements from the water column surface to just above the bottom (~880 meters). Alongside CTD data, we would capture seawater at each site, including the surface, deep chlorophyll maximum, oxycline (~150 m), and hypoxic depth (~880 m). By collecting this seawater monthly, we have been able to capture seasonal and annual trends in the microbial community (i.e., species richness, relative abundances).</p>
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/SPOT-vert-profile.png" class="img-fluid figure-img" style="width:20.0%"></p>
<p></p><figcaption class="figure-caption">Seasonal depth profiles from the SPOT station from Hu et al.&nbsp;2016</figcaption><p></p>
</figure>
</div>
</div>
</div>
<p><em>Findings from the SPOT station</em></p>
<blockquote class="blockquote">
<p>The oxycline hosts increased microbial activity.</p>
</blockquote>
<p>Naturally-occurring transition zones, such as nutriclines or oxyclines (where oxygen dramatically decreases) attract microorganisms. As a result, there is typically an uptick in microbial species richness, abundance, and activity localized to these sites. Subsequently, microorganisms that consume other microbes (such as protistan heterotrophs) will also be draw to this environment with available prey.</p>
<p>DNA-based observations of microbial eukaryotic populations had revealed that heterotrophic species where present throughout the water column and had a relative increase in the number of species at the oxycline. In a paper from <a href="https://academic.oup.com/femsec/article/92/4/fiw050/2197988">2016</a>, we used both RNA and DNA tag-sequencing analyses to (1) assess if there was a difference in using RNA and DNA in tag-sequence surveys and to (2) determine if this approach could be used to infer metabolic activity.</p>
<p>To access the metabolically-active component of the protistan community at SPOT, we also <a href="https://sfamjournals.onlinelibrary.wiley.com/doi/abs/10.1111/1462-2920.14259">applied metatranscriptomics</a>. After extracting total RNA from each depth at the SPOT station, we isolated and sequenced the messenger RNA. By matching mRNA sequences to reference transcriptome databases, we can annotate the taxonomic origin and gene identity (functional potential). In the case of the metabolically diverse protists, results from the metatranscriptome analysis brought us closer to simultaneously characterizing the trophic mode and taxonomic identity. At the oxycline, there was a distinct shift in the known heterotrophic taxa, such as ciliates - where a relative increase in transcripts related to fatty acid breakdown was observed. Similarly, taxa known to be mixotrophic (e.g., haptophytes) had distinct transcript profiles in the euphotic and sub-euphotic zones. At the surface, transcripts associated with phototrophy were higher compared to transcripts associated with heterotrophic modes of nutrition at depths without sunlight.</p>
<blockquote class="blockquote">
<p>Surface phytoplankton community varies according to environment.</p>
</blockquote>
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/IMG-4423.JPG" class="img-fluid figure-img" style="width:40.0%"></p>
<p></p><figcaption class="figure-caption">HOT cruise</figcaption><p></p>
</figure>
</div>
</div>
</div>
</section>
<section id="hot" class="level2">
<h2 class="anchored" data-anchor-id="hot">HOT</h2>
<p>The <a href="http://aco-ssds.soest.hawaii.edu/ALOHA/">Hawaii Ocean Time-series (HOT)</a> samples station ALOHA - or A Long-Term Oligotrophic Habitat Assessment monthly. This time series has been running for over 30 years. As part of the Simons Foundation funded SCOPE collaboration, I was able to be involved in several cruises to station ALOHA. This was an exciting opportunity to study the oligotrophic North Pacific Subtropical Gyre and compare our findings and methods to the coastal ecosystem of California.</p>
<p>For one field season, we conducted high-resolution diel sampling by following a parcel of water and sampling every 4 hours. As a result of this work, we captured daily fluctuations in microbial community dynamics, including species compositions, microbe-microbe-interactions, grazing activity, and details of how microbial photosynthetic machinery operates on an hourly time scale.</p>
<blockquote class="blockquote">
<p>Species-specific nutritional needs drive temporal niche partitioning among protistan primary producers and consumers.</p>
</blockquote>
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="projects_files/figure-html/unnamed-chunk-19-1.png" class="img-fluid figure-img" width="768"></p>
</figure>
</div>
</div>
</div>
</section>
</section>
<section id="data-science" class="level1">
<h1>Data science</h1>
<p>Despite their ecological importance, efforts to catalog protistan physiological abilities and trophic strategies lag behind those of their prokaryotic counterparts. The interpretation of large sequence datasets relies on the knowledge gleaned from culture-based transcriptome studies, new developments in genetic probing of microorganisms, and single-cell sequencing. It is imperative that we build reproducible pipelines and infrastructure to keep up with growing sequence information.</p>
<p>My two data science related goals are to (1) streamline the methods to link genotype and phenotype for environmentally-relevant microbial eukaryotes and (2) create an engaging environment to teach the next generation of scholars methods in computational approaches.</p>
<p>Contact me if you are interested in building up a ‘data science community’.</p>
<section id="reproducible-pipelines-bioinformatic-tools" class="level2">
<h2 class="anchored" data-anchor-id="reproducible-pipelines-bioinformatic-tools">Reproducible pipelines &amp; Bioinformatic tools</h2>
<ul>
<li><p><a href="https://github.com/alexanderlabwhoi/EukHeist">EUKHeist</a> &amp; <a href="https://www.biorxiv.org/content/10.1101/2021.07.25.453713v1">preprint</a></p></li>
<li><p><a href="https://eukulele.readthedocs.io/en/latest/">EUKulele</a></p></li>
<li><p><a href="https://github.com/shu251/tagseq-qiime2-snakemake">Tag-sequencing work flows</a></p></li>
<li><p><a href="https://github.com/shu251/Extract_microeukrRNA_from_metagenome">Extracting microbial eukaryotic information from metagenomes</a></p></li>
<li><p><a href="https://github.com/shu251/analyzing_microbiome_timeseries">Analysing microbiome time-series</a></p></li>
</ul>


</section>
</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    target: function(trigger) {
      return trigger.previousElementSibling;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    setTimeout(function() {
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const cites = ref.parentNode.getAttribute('data-cites').split(' ');
    tippyHover(ref, function() {
      var popup = window.document.createElement('div');
      cites.forEach(function(cite) {
        var citeDiv = window.document.createElement('div');
        citeDiv.classList.add('hanging-indent');
        citeDiv.classList.add('csl-entry');
        var biblioDiv = window.document.getElementById('ref-' + cite);
        if (biblioDiv) {
          citeDiv.innerHTML = biblioDiv.innerHTML;
        }
        popup.appendChild(citeDiv);
      });
      return popup.innerHTML;
    });
  }
});
</script>
</div> <!-- /content -->



</body></html>