<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.3.353">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>Data science</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
</style>


<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit"
  }
}</script>


<link rel="stylesheet" href="styles.css">
</head>

<body class="floating nav-fixed slimcontent">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg navbar-dark ">
      <div class="navbar-container container-fluid">
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll ms-auto">
  <li class="nav-item">
    <a class="nav-link" href="./index.html" rel="" target="">
 <span class="menu-text">Hu lab</span></a>
  </li>  
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-about" role="button" data-bs-toggle="dropdown" aria-expanded="false" rel="" target="">
 <span class="menu-text">about</span>
    </a>
    <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="nav-menu-about">    
        <li>
    <a class="dropdown-item" href="./team.html" rel="" target="">
 <span class="dropdown-text">Lab members</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="./lab-policies.html" rel="" target="">
 <span class="dropdown-text">Lab community</span></a>
  </li>  
    </ul>
  </li>
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-research" role="button" data-bs-toggle="dropdown" aria-expanded="false" rel="" target="">
 <span class="menu-text">research</span>
    </a>
    <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="nav-menu-research">    
        <li>
    <a class="dropdown-item" href="./deepsea-vents.html" rel="" target="">
 <span class="dropdown-text">Deep-sea vents</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="./time-series.html" rel="" target="">
 <span class="dropdown-text">Time-series</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="./data-science.html" rel="" target="">
 <span class="dropdown-text">Data science</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="./publications.html" rel="" target="">
 <span class="dropdown-text">Publications</span></a>
  </li>  
    </ul>
  </li>
</ul>
            <div class="quarto-navbar-tools tools-wide">
    <a href="https://github.com/shu251" title="" class="quarto-navigation-tool px-1" aria-label=""><i class="bi bi-github"></i></a>
    <a href="https://www.instagram.com/microhukaryote/" title="" class="quarto-navigation-tool px-1" aria-label=""><i class="bi bi-instagram"></i></a>
    <a href="https://scholar.google.com/citations?user=9CI91CMAAAAJ&amp;hl=en" title="" class="quarto-navigation-tool px-1" aria-label=""><i class="bi bi-share"></i></a>
</div>
          </div> <!-- /navcollapse -->
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal sidebar-navigation floating overflow-auto">
    <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#computational-biology" id="toc-computational-biology" class="nav-link active" data-scroll-target="#computational-biology"><span class="header-section-number">1</span> Computational biology</a>
  <ul class="collapse">
  <li><a href="#tag-sequencing" id="toc-tag-sequencing" class="nav-link" data-scroll-target="#tag-sequencing"><span class="header-section-number">1.1</span> Tag-sequencing</a></li>
  <li><a href="#time-series-data" id="toc-time-series-data" class="nav-link" data-scroll-target="#time-series-data"><span class="header-section-number">1.2</span> Time-series data</a></li>
  <li><a href="#metatranscriptomics" id="toc-metatranscriptomics" class="nav-link" data-scroll-target="#metatranscriptomics"><span class="header-section-number">1.3</span> Metatranscriptomics</a>
  <ul class="collapse">
  <li><a href="#computational-tools" id="toc-computational-tools" class="nav-link" data-scroll-target="#computational-tools"><span class="header-section-number">1.3.1</span> Computational tools</a></li>
  <li><a href="#building-reproducible-pipeline-for-amplicon-sequence-analysis" id="toc-building-reproducible-pipeline-for-amplicon-sequence-analysis" class="nav-link" data-scroll-target="#building-reproducible-pipeline-for-amplicon-sequence-analysis"><span class="header-section-number">1.3.2</span> Building reproducible pipeline for amplicon sequence analysis</a></li>
  <li><a href="#metaeukomic" id="toc-metaeukomic" class="nav-link" data-scroll-target="#metaeukomic"><span class="header-section-number">1.3.3</span> Metaeukomic</a></li>
  </ul></li>
  </ul></li>
  <li><a href="#database-darkly" id="toc-database-darkly" class="nav-link" data-scroll-target="#database-darkly"><span class="header-section-number">2</span> Database Darkly</a></li>
  <li><a href="#r-tutorials-workshops" id="toc-r-tutorials-workshops" class="nav-link" data-scroll-target="#r-tutorials-workshops"><span class="header-section-number">3</span> R tutorials &amp; workshops</a></li>
  <li><a href="#r-packages-builds" id="toc-r-packages-builds" class="nav-link" data-scroll-target="#r-packages-builds"><span class="header-section-number">4</span> R packages &amp; builds</a>
  <ul class="collapse">
  <li><a href="#axial-seamount-wordle" id="toc-axial-seamount-wordle" class="nav-link" data-scroll-target="#axial-seamount-wordle"><span class="header-section-number">4.1</span> Axial Seamount Wordle</a></li>
  <li><a href="#woods-hole-color-palette" id="toc-woods-hole-color-palette" class="nav-link" data-scroll-target="#woods-hole-color-palette"><span class="header-section-number">4.2</span> Woods Hole color palette</a></li>
  <li><a href="#exercise" id="toc-exercise" class="nav-link" data-scroll-target="#exercise"><span class="header-section-number">4.3</span> ExeRcise</a></li>
  <li><a href="#lab-work" id="toc-lab-work" class="nav-link" data-scroll-target="#lab-work"><span class="header-section-number">4.4</span> Lab work</a></li>
  </ul></li>
  </ul>
</nav>
</nav>
<div id="quarto-sidebar-glass" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
    </div>
<!-- main -->
<main class="content page-columns page-full" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Data science</h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  

</header>

<p>Despite their ecological importance, efforts to catalog protistan physiological abilities and trophic strategies lag behind those of their prokaryotic counterparts. The interpretation of large sequence datasets relies on the knowledge gleaned from culture-based transcriptome studies, new developments in genetic probing of microorganisms, and single-cell sequencing. It is imperative that we build reproducible pipelines and infrastructure to keep up with growing sequence information.</p>
<section id="computational-biology" class="level1 page-columns page-full" data-number="1">
<h1 data-number="1"><span class="header-section-number">1</span> Computational biology</h1>

<div class="no-row-height column-margin column-container"><div class="">
<p><img src="images/Hu Lab Logo Final-01.png" class="img-fluid"></p>
</div></div><section id="tag-sequencing" class="level2" data-number="1.1">
<h2 data-number="1.1" class="anchored" data-anchor-id="tag-sequencing"><span class="header-section-number">1.1</span> Tag-sequencing</h2>
<p>To address questions related to microeukaryotic biodiversity, we use 18S rRNA gene tag-sequencing (also called metabarcoding). Below are resources for analyzing this type of data.</p>
<p>Our most recent <a href="https://github.com/shu251/qiime2-2024">tag-sequencing work flow</a> was updated in 2024.</p>
<p>Check my <a href="https://github.com/shu251">github</a> for <em>in process</em> work, where we are always working to update our code.</p>
<p>Also see the <a href="https://biovcnet.github.io">BVCN</a> resources for amplicon sequencing</p>
</section>
<section id="time-series-data" class="level2" data-number="1.2">
<h2 data-number="1.2" class="anchored" data-anchor-id="time-series-data"><span class="header-section-number">1.2</span> Time-series data</h2>
<p>When we work with any metabarcoding data, it is important to acknowledge and analyze it as <em>compositional</em> data. As part of our R code clubs we always emphasize the inclusion of diagnostic steps to make sure our data is being analyzed appropriately.</p>
<p>A product from a series of discussions with colleagues is <a href="https://github.com/shu251/analyzing_microbiome_timeseries">resource for how to analyze time-series data.</a>. When paired with the <a href="https://www.frontiersin.org/journals/genetics/articles/10.3389/fgene.2020.00310/full">publication</a> it is a good reminder and starting point when analyzing this type of dataset.</p>
<p>Additional resources include the <a href="https://github.com/shu251/18Sdiversity_diel">code</a> and publication, called <a href="https://www.frontiersin.org/articles/10.3389/fmars.2018.00351/full">A Hard Day’s Night</a>, which explores temporal niche partitioning among Station ALOHA microeukaryotic communities.</p>
<div class="quarto-figure quarto-figure-left">
<figure class="figure">
<p><img src="images/timeseries.jpg" class="lightbox img-fluid figure-img" width="300"></p>
<figcaption class="figure-caption">Figure 2 in Coenen et al.&nbsp;2020</figcaption>
</figure>
</div>
</section>
<section id="metatranscriptomics" class="level2 page-columns page-full" data-number="1.3">
<h2 data-number="1.3" class="anchored" data-anchor-id="metatranscriptomics"><span class="header-section-number">1.3</span> Metatranscriptomics</h2>
<p>To interrogate <em>in situ</em> microeukaryotic metabolic activity, we can target the messengar RNA (mRNA) for sequencing through metatranscriptomics.</p>
<p>Our most recent <a href="https://shu251.github.io/meta-eukomic/main.html">introductory tutorial on metatranscriptomics, especially for eukaryotes, can be found here</a>.</p>
<section id="computational-tools" class="level3 page-columns page-full" data-number="1.3.1">
<h3 data-number="1.3.1" class="anchored" data-anchor-id="computational-tools"><span class="header-section-number">1.3.1</span> Computational tools</h3>
<p>An active area of our research includes developing solutions to the computational challenges presented by eukaryotic genetics, a part of this involves creating tools to automate the assembly, annotation, and processing of meta’omic (metagenome/metatranscriptome) data (EUKulele, EUKHeist) and augment how sequence reference databases are queried (see Krinos et al.&nbsp;2024).</p>
<p><a href="https://github.com/alexanderlabwhoi/EukHeist">EUKHeist code</a> and <a href="https://journals.asm.org/doi/full/10.1128/mbio.01676-23?rfr_dat=cr_pub++0pubmed&amp;url_ver=Z39.88-2003&amp;rfr_id=ori%3Arid%3Acrossref.org">publication</a></p>

<div class="no-row-height column-margin column-container"><div class="">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/eukheist.jpg" class="img-fluid figure-img"></p>
<figcaption class="figure-caption">Alexander et al.&nbsp;2023</figcaption>
</figure>
</div>
</div></div><p><a href="https://eukulele.readthedocs.io/en/latest/">EUKulele documentation</a> and <a href="https://joss.theoj.org/papers/10.21105/joss.02817">publication</a>.</p>
</section>
<section id="building-reproducible-pipeline-for-amplicon-sequence-analysis" class="level3" data-number="1.3.2">
<h3 data-number="1.3.2" class="anchored" data-anchor-id="building-reproducible-pipeline-for-amplicon-sequence-analysis"><span class="header-section-number">1.3.2</span> Building reproducible pipeline for amplicon sequence analysis</h3>
<p>Instructions for implementing a snakemake workflow to make tag-sequencing pipeline more robust <a href="https://github.com/shu251/tagseq-qiime2-snakemake">HERE</a></p>
<p><a href="https://github.com/shu251/db-build-microeuks">Building/prepping</a> a database for use in QIIME2</p>
</section>
<section id="metaeukomic" class="level3" data-number="1.3.3">
<h3 data-number="1.3.3" class="anchored" data-anchor-id="metaeukomic"><span class="header-section-number">1.3.3</span> Metaeukomic</h3>
<p>Members of the Hu lab are contributing to community-wide efforts to examine the range of variability across metatranscriptome bench and computational pipelines. This an intercalibration effort funded by OCB, <a href="https://www.us-ocb.org/metat-intercomparison/">Metaeukomic</a>.</p>
</section>
</section>
</section>
<section id="database-darkly" class="level1 page-columns page-full" data-number="2">
<h1 data-number="2"><span class="header-section-number">2</span> Database Darkly</h1>

<div class="no-row-height column-margin column-container"><div class="">
<p><img src="images/db_darkly_logo_emptybg.png" class="img-fluid"></p>
</div></div><p>A challenge we face with deep-sea biodiversity work is the lack of available genetic reference databases. Using results from a <a href="https://onlinelibrary.wiley.com/doi/abs/10.1111/mec.16745">large tag-sequencing survey</a>, a team of undergraduate researchers took a deep dive into mining biological and ecological information about deep-sea protistan species.</p>
<p><a href="https://shu251.github.io/deep-sea-microeuk-DB/">Visit the Database Darkly website</a> to learn more about the microeukaryotes that live in the dark ocean.</p>
<p><img src="images/OCNG-symposium-UndergraduateResearch.png" class="img-fluid">{.lightbox,fig-align=“center”, width=“450”}</p>
</section>
<section id="r-tutorials-workshops" class="level1" data-number="3">
<h1 data-number="3"><span class="header-section-number">3</span> R tutorials &amp; workshops</h1>
<ul>
<li><p><a href="https://shu251.github.io/r-ventures/#introduction">R code examples and tutorials</a></p></li>
<li><p><a href="https://github.com/shu251/PreliminaryFigures_V4_tagseq">Generating figures</a> from HTS tag-sequencing data - OTU or ASV table as input</p></li>
<li><p>Learning R with <a href="https://biovcnet.github.io">BVCN</a></p></li>
<li><p><a href="https://github.com/shu251/Extract_microeukrRNA_from_metagenome">Extracting microbial eukaryotic information from metagenomes</a></p></li>
</ul>
</section>
<section id="r-packages-builds" class="level1" data-number="4">
<h1 data-number="4"><span class="header-section-number">4</span> R packages &amp; builds</h1>
<section id="axial-seamount-wordle" class="level2" data-number="4.1">
<h2 data-number="4.1" class="anchored" data-anchor-id="axial-seamount-wordle"><span class="header-section-number">4.1</span> Axial Seamount Wordle</h2>
<p><a href="https://sarah-k-hu.shinyapps.io/wordle-axial/">Play our custom Axial Seamount-themed wordle</a></p>
<p>Part of our <a href="https://www2.whoi.edu/staff/jhuber/protatax2022/">#protatax</a> expedition.</p>
<div class="cell">
<div class="cell-output-display">
<p><img src="images/vent-logo-protatax_JulieForCruise.jpg" class="img-fluid" style="width:50.0%"></p>
</div>
</div>
</section>
<section id="woods-hole-color-palette" class="level2" data-number="4.2">
<h2 data-number="4.2" class="anchored" data-anchor-id="woods-hole-color-palette"><span class="header-section-number">4.2</span> Woods Hole color palette</h2>
<p>Download color palettes inspired by Cape Cod and Woods Hole area. Includes docks, sunsets, WHOI logo, and RV Atlantis. <a href="https://github.com/shu251/PaletteWoodsHole">github</a></p>
<div class="cell">
<div class="cell-output-display">
<p><img src="images/knob1.png" class="img-fluid" style="width:50.0%"></p>
</div>
</div>
</section>
<section id="exercise" class="level2" data-number="4.3">
<h2 data-number="4.3" class="anchored" data-anchor-id="exercise"><span class="header-section-number">4.3</span> ExeRcise</h2>
<p>Generate a randomized custom workout. Inspired by High Intensity Interval training workouts, user defines warm up and workout interval length and total number of exercises. Additional functions to run workout with a timer in R and a tabata-style workout. <a href="https://github.com/shu251/exRcise">github</a></p>
<hr>
</section>
<section id="lab-work" class="level2" data-number="4.4">
<h2 data-number="4.4" class="anchored" data-anchor-id="lab-work"><span class="header-section-number">4.4</span> Lab work</h2>
<p>Built <em>for personal use only</em>. List of functions to streamline molecular bench work and common math required in preparing genetic material for sequencing. e.g., PCR master mix calculation, CV=CV, and RNA quantification preparation steps. <a href="https://github.com/shu251/lab">github repo</a></p>
<div class="cell">
<div class="cell-output-display">
<p><img src="images/knob.jpg" class="img-fluid" style="width:50.0%"></p>
</div>
</div>
<p><em>the Knob in Woods Hole, MA</em></p>


</section>
</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->



</body></html>