<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.3.353">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>Publications</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
</style>


<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit"
  }
}</script>


<link rel="stylesheet" href="styles.css">
</head>

<body class="nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg navbar-dark ">
      <div class="navbar-container container-fluid">
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll ms-auto">
  <li class="nav-item">
    <a class="nav-link" href="./index.html" rel="" target="">
 <span class="menu-text">Hu lab</span></a>
  </li>  
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-about" role="button" data-bs-toggle="dropdown" aria-expanded="false" rel="" target="">
 <span class="menu-text">About</span>
    </a>
    <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="nav-menu-about">    
        <li>
    <a class="dropdown-item" href="./team.html" rel="" target="">
 <span class="dropdown-text">Lab members</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="./lab-policies.html" rel="" target="">
 <span class="dropdown-text">Lab community</span></a>
  </li>  
    </ul>
  </li>
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-research" role="button" data-bs-toggle="dropdown" aria-expanded="false" rel="" target="">
 <span class="menu-text">Research</span>
    </a>
    <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="nav-menu-research">    
        <li>
    <a class="dropdown-item" href="./projects.html" rel="" target="">
 <span class="dropdown-text">Research themes</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="./deepsea-vents.html" rel="" target="">
 <span class="dropdown-text">Deep-sea vents</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="./time-series.html" rel="" target="">
 <span class="dropdown-text">Time-series</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="./data-science.html" rel="" target="">
 <span class="dropdown-text">Data science</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="./publications.html" rel="" target="">
 <span class="dropdown-text">Publications</span></a>
  </li>  
    </ul>
  </li>
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-resources" role="button" data-bs-toggle="dropdown" aria-expanded="false" rel="" target="">
 <span class="menu-text">Resources</span>
    </a>
    <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="nav-menu-resources">    
        <li>
    <a class="dropdown-item" href="./resources.html" rel="" target="">
 <span class="dropdown-text">Tools</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="./r-packages.html" rel="" target="">
 <span class="dropdown-text">R packages &amp; projects</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="https://shu251.github.io/r-ventures/#introduction" rel="" target="">
 <span class="dropdown-text">Learn R</span></a>
  </li>  
    </ul>
  </li>
</ul>
            <div class="quarto-navbar-tools tools-wide">
    <a href="https://github.com/shu251" title="" class="quarto-navigation-tool px-1" aria-label=""><i class="bi bi-github"></i></a>
    <a href="https://www.instagram.com/microhukaryote/" title="" class="quarto-navigation-tool px-1" aria-label=""><i class="bi bi-instagram"></i></a>
    <a href="https://scholar.google.com/citations?user=9CI91CMAAAAJ&amp;hl=en" title="" class="quarto-navigation-tool px-1" aria-label=""><i class="bi bi-share"></i></a>
</div>
          </div> <!-- /navcollapse -->
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar"><div class="quarto-margin-header"><div class="margin-header-item">
<p><img src="./images/logo-hu.png" class="img-fluid"></p>
</div></div>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Publications</h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  

</header>

<section id="pending" class="level1">
<h1><em>pending</em></h1>
<ul>
<li><p>Gleich, S. J., Mesrop, L. Y., Cram, J. A., Weissman, J. L., Hu, S. K., Yeh, Y., Fuhrman, J. A., &amp; Caron, D. A. (2024) With a little help from my friends: Importance of protist-protist interactions in structuring marine protistan communities in the San Pedro Channel. (Accepted at mSystems)</p></li>
<li><p>Hu, S.K., Anderson, R.A, Serres M.H., Sylva S.P., German, C.R., Seewald, J.S., Lang, S.Q., &amp; Huber, J.A. Exploration of ciliate life styles in deep sea hydrothermal vents. (<em>In Prep</em>).</p></li>
<li><p>Neely, C.J., Hu, S.K., Alexander, H., Tully, B.J. The high-throughput gene prediction of more than 1,700 eukaryote genomes using the software package EukMetaSanity. (<em>In Prep</em>) <a href="https://www.biorxiv.org/content/10.1101/2021.07.25.453296v1">bioRxiv: 2021.07.25.453296</a></p></li>
<li><p>Hu, S.K., Alexander, H., Liu, Z., Heidelberg, K.B., Dyhrman, S. &amp; Caron, D.A. Distinct transcriptional signatures of the protistan community at the SPOT and ALOHA ocean time-series stations. (<em>In Prep</em>).</p></li>
</ul>
</section>
<section id="section" class="level1">
<h1>2024</h1>
<ul>
<li><p>Krinos, A.I., Mars Brisbin, M., Hu, S.K., Cohen N.R., Rynearson, T.A., Follows, M.J., Schulz, F., &amp; Alexander, H. (2024) Missing Microbial Eukaryotes and Misleading Meta-Omic Conclusions. Nature Communications; 15 (1): 9873. https://doi.org/10.1038/s41467-024-52212-w.</p></li>
<li><p>Hu S. K., Anderson R.E., Pachiadaki M.G., Edgcomb V.P., Serres M.H., Sylva S.P., German, C.R., Seewald, J.S., Lang, S.Q., &amp; Huber, J.A. (2024). Microbial eukaryotic predation pressure and biomass at deep-sea hydrothermal vents. ISME Journal; 18.</p></li>
<li><p>Gleich S.J., Hu S.K., Krinos A.I., &amp; Caron D.A. (2024) Protistan community composition and metabolism in the North Pacific Subtropical Gyre: Influences of mesoscale eddies and depth. Environmental Microbiology; 26: e16556.</p></li>
<li><p>Beckett S.J., Demory D., Coenen A.R., Casey J.R., Dugenne M., Follett C.L., Connell P., Carlson M.C.G., Hu S.K., &amp; 12 others. (2024) Disentangling top-down drivers of mortality underlying diel population dynamics of Prochlorococcus in the North Pacific Subtropical Gyre. Nature Communications; 15: 2105.</p></li>
</ul>
</section>
<section id="section-1" class="level1">
<h1>2023</h1>
<ul>
<li><p>Ollison, G.A., Hu, S.K., Hopper, J.V., Stewart, B.P., Beatty, J.L., &amp; Caron, D.A. (2023). Physiology governing diatom vs.&nbsp;dinoflagellate bloom and decline in coastal Santa Monica Bay. Frontiers in Microbiology; 14. https://doi.org/10.3389/fmicb.2023.1287326</p></li>
<li><p>Alexander, H., Hu, S.K., Krinos, A.I., Pachiadaki, M., Tully, B.J., Neely, C.J., &amp; Reiter, T. (2023) Eukaryotic genomes from a global metagenomic data set illuminate trophic modes and biogeography of ocean plankton. mBio; e0167623. DOI: 10.1128/mbio.01676-23</p></li>
</ul>
</section>
<section id="section-2" class="level1">
<h1>2022</h1>
<ul>
<li><p>Hu, S. K., Smith, A. R., Anderson, R. E., Sylva, S. P., Setzer, M., Steadmon, M., Frank, K. L., Chan, E. W., Lim, D. S. S., German, C. R., Breier, J. A., Lang, S. Q., Butterfield, D. A., Fortunato, C. S., Seewald, J. S., &amp; Huber, J. A. (2022). Globally-distributed microbial eukaryotes exhibit endemism at deep-sea hydrothermal vents. Molecular Ecology, 00, 1–19. https://doi.org/10.1111/mec.16745 <a href="https://doi.org/10.1111/mec.16745">link</a> <a href="https://shu251.github.io/microeuk-amplicon-survey/">code</a></p></li>
<li><p>Hammond, S.W,, Lodolo, L., Hu, S.K., &amp; Pasulka, A.L. (2022) Methodological ‘lenses’ influence the characterization of phytoplankton dynamics in a coastal upwelling ecosystem. Environmental Microbiology Reports <a href="https://doi.org/10.1111/1758-2229.13116">link</a></p></li>
<li><p>Ollison, G.A., Hu, S.K., Hopper, J.V., Stewart, B.P., Smith, J., Beatty, J.L., et al.&nbsp;(2022) Daily dynamics of contrasting spring algal blooms in Santa Monica Bay (central Southern California Bight). Environmental Microbiology 1462-2920.16137. <a href="https://doi.org/10.1111/1462-2920.16137">link</a></p></li>
<li><p>Cohen, N.R., Alexander, H., Krinos, A.I., Hu, S.K., and Lampe, R.H. (2022) Marine Microeukaryote Metatranscriptomics: Sample Processing and Bioinformatic Workflow Recommendations for Ecological Applications. Front Mar Sci 9: 867007. <a href="https://www.frontiersin.org/articles/10.3389/fmars.2022.867007/full">link</a></p></li>
</ul>
</section>
<section id="section-3" class="level1">
<h1>2021</h1>
<ul>
<li><p>Tully, B.J., Buongiorno, J., Cohen, A.B., Cram, J.A., Garber, A.I., Hu, S.K., et al.. The Bioinformatics Virtual Coordination Network: an open-source and interactive learning environment. (2021) Frontiers in Education: STEM Education. <a href="https://www.frontiersin.org/articles/10.3389/feduc.2021.711618/full">link</a></p></li>
<li><p>Ollison G., Hu, S.K., Mesrop, L.Y., Delong, E. &amp; Caron, D. A. (2021) Come Rain or Shine: Depth Not Season Shapes the Protistan Community at Station ALOHA in the North Pacific Subtropical Gyre. Deep-Sea Research Part II. <a href="https://www.sciencedirect.com/science/article/pii/S0967063721000339">link</a></p></li>
<li><p>Hu, S.K., Herrera, E.L., Smith, A.R., Pachiadaki, M.G., Edgcomb, V.P., Sylva, S.P., Chan E.W., Seewald, J.S., German, C.R., &amp; Huber, J.A. (2021) Protistan grazing impacts microbial communities and carbon cycling in the deep-sea hydrothermal vent environment. Proc Natl Acad Sci USA <a href="https://www.pnas.org/content/118/29/e2102674118">link</a> <a href="https://shu251.github.io/protist-gordaridge-2021/">code</a></p></li>
</ul>
</section>
<section id="section-4" class="level1">
<h1>2020</h1>
<ul>
<li><p>Coesel, S.N., Durham, B.P., Groussman, R.D., Hu, S.K., Caron, D.A., Morales, R.L., et al.&nbsp;(2021) Diel transcriptional oscillations of light-sensitive regulatory elements in open-ocean eukaryotic plankton communities. Proc Natl Acad Sci USA 118: 12. <a href="https://www.pnas.org/content/118/6/e2011038118">link</a></p></li>
<li><p>Krinos, A., Hu, S., Cohen, N., and Alexander, H. (2021) EUKulele: Taxonomic annotation of the unsung eukaryotic microbes. JOSS 6: 2817. <a href="https://joss.theoj.org/papers/10.21105/joss.02817">link</a></p></li>
<li><p>Goordial, J., Hu, S., &amp; Tully, B. (2020). C-DEBI NextGen 2019 Early Career perspective on ‘What’s Next?’: Upcoming Challenges and Opportunities. DOI: 10.31219/osf.io/7xkpq <a href="https://www.darkenergybiosphere.org/publication/c-debi-nextgen-2019-early-career-perspective-on-whats-next-upcoming-challenges-and-opportunities/">link</a></p></li>
<li><p>Coenen, A.R., Hu, S.K., Luo, E., Muratore, D., and Weitz, J.S. (2020) A Primer for Microbiome Time-Series Analysis. Front Genet 11: 310. <a href="https://www.frontiersin.org/articles/10.3389/fgene.2020.00310/full">link</a></p></li>
<li><p>Lim, Darlene S.S., Raineault, N.A., Brier, J.A., Chan, E., Chernov, J., Cohen, T., Deans, M., Garcia, A., German, C. R., Hauer, M., Hu, S.K., Huber, J.A., Kane, R., Kobs Nawotniak, S., Lees, D., Lowe, J., Lubetkin, M., Marsh, L., Milesi, V., Miller, M., Miramalek, Z., Saunders, M., Sharif, K., Shields, A., Shock, E., Smith, A.R., and Sylva, S. (2020), SUBSEA 2019 Expedition to the Gorda Ridge, Oceanography 33(1), Supplement Pages 36 – 37. <a href="https://scholarworks.utrgv.edu/cgi/viewcontent.cgi?article=1067&amp;context=eems_fac">link</a></p></li>
</ul>
</section>
<section id="section-5" class="level1">
<h1>2019</h1>
<ul>
<li><p>Boeuf, D., Edwards, B.R., Eppley, J.M., Hu, S.K., Poff, K.E., Romano, A.E., et al.&nbsp;(2019) Biological composition and microbial dynamics of sinking particulate organic matter at abyssal depths in the oligotrophic open ocean. Proc Natl Acad Sci USA 11824–11832. <a href="https://www.pnas.org/content/116/24/11824">link</a></p></li>
<li><p>Liu, Z., Mesrop, L.Y., Hu, S.K., and Caron, D.A. (2019) Transcriptome of <em>Thalassicolla nucleata</em> Holobiont Reveals Details of a Radiolarian Symbiotic Relationship. Front Mar Sci 6: 284. <a href="https://www.frontiersin.org/articles/10.3389/fmars.2019.00284/full">link</a></p></li>
<li><p>Pasulka, A., Hu, S.K., Countway, P.D., Coyne, K.J., Cary, S.C., Heidelberg, K.B., and Caron, D.A. (2019) SSU rRNA Gene Sequencing Survey of Benthic Microbial Eukaryotes from Guaymas Basin Hydrothermal Vent. Journal of Eukaryotic Microbiology jeu.12711.<a href="https://onlinelibrary.wiley.com/doi/10.1111/jeu.12711">link</a></p></li>
<li><p>Caron, D.A. and Hu, S.K. (2019) Are We Overestimating Protistan Diversity in Nature? Trends in Microbiology 27: 197–205.<a href="https://www.sciencedirect.com/science/article/abs/pii/S0966842X18302324?via%3Dihub">link</a></p></li>
</ul>
</section>
<section id="section-6" class="level1">
<h1>2018</h1>
<ul>
<li><p>Hu, S.K., Liu, Z., Alexander, H., Campbell, V., Connell, P.E., Dyhrman, S.T., et al.&nbsp;(2018) Shifting metabolic priorities among key protistan taxa within and below the euphotic zone: Depth-related protistan metatranscriptomes. Environ Microbiol 20: 2865–2879. <a href="https://sfamjournals.onlinelibrary.wiley.com/doi/abs/10.1111/1462-2920.14259">link</a></p></li>
<li><p>Hu, S.K., Paige E. Connell, Mesrop, L.Y., and Caron, D.A. (2018) A Hard Day’s Night: Diel Shifts in Microbial Eukaryotic Activity in the North Pacific Subtropical Gyre. Front Mar Sci 5:251. <a href="https://dornsife.usc.edu/assets/sites/378/docs/Caron_pdfs/2018_Hu_etal_FrontiersMarineScience.pdf">link</a></p></li>
</ul>
</section>
<section id="section-7" class="level1">
<h1>2017</h1>
<ul>
<li><p>Liu, Z., Hu, S.K., Campbell, V., Tatters, A.O., Heidelberg, K.B., and Caron, D.A. (2017) Single-cell transcriptomics of small microbial eukaryotes: limitations and potential. The ISME Journal 11: 1282–1285. <a href="https://www.nature.com/articles/ismej2016190">link</a></p></li>
<li><p>Connell, P.E., Campbell, V., Gellene, A.G., Hu, S.K., and Caron, D.A. (2017) Planktonic food web structure at a coastal time-series site: II. Spatiotemporal variability of microbial trophic activities. Deep Sea Res Part I Oceanogr Res Pap 121: 210–223. <a href="https://doi.org/10.1016/j.dsr.2017.01.007">link</a></p></li>
</ul>
</section>
<section id="section-8" class="level1">
<h1>2016</h1>
<ul>
<li>Hu, S.K., Campbell, V., Connell, P., Gellene, A.G., Liu, Z., Terrado, R., and Caron, D.A. (2016) Protistan diversity and activity inferred from RNA and DNA at a coastal ocean site in the eastern North Pacific. FEMS Microbiol Ecol fiw050. <a href="https://academic.oup.com/femsec/article/92/4/fiw050/2197988">link</a></li>
</ul>
</section>
<section id="section-9" class="level1">
<h1>2015</h1>
<ul>
<li><p>Hu, S.K., Liu, Z., Lie, A.A.Y., Countway, P.D., Kim, D.Y., Jones, A.C., et al.&nbsp;(2015) Estimating Protistan Diversity Using High-Throughput Sequencing. Journal of Eukaryotic Microbiology 62: 688-693. <a href="https://doi.org/10.1111/jeu.12217">link</a></p></li>
<li><p>Lie, A.A.Y., Liu, Z., Hu, S.K., Jones, A.C., Kim, D.Y., Countway, P.D., et al.&nbsp;(2014) Investigating Microbial Eukaryotic Diversity from a Global Census: Insights from a Comparison of Pyrotag and Full-Length Sequences of 18S rRNA Genes. Applied and Environmental Microbiology 80: 4363-4373. <a href="https://doi.org/10.1128/AEM.00057-14">link</a></p></li>
</ul>


</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->



</body></html>